tests/bibo/test_query.py

```diff
@@ -49,10 +49,58 @@ def test_search_with_capitalized_search_term(data):
     """Issue #28"""
     results = list(query.search(data, ["ASIMOV"]))
     assert len(results) == 1
-    assert results[0]["fields"]["title"] == "Foundation"
+    assert results[0].entry["fields"]["title"] == "Foundation"
 
 
 def test_open_multiple_entries_one_exact_match(data):
     with pytest.raises(click.ClickException):
         query.get(data, ["ab"])
     query.get(data, ["abc"])
+
+
+def test_search_match_details(data):
+    results = list(query.search(data, ["tolkien", "hobbit"]))
+    assert len(results) == 1
+    assert "tolkien" in results[0].match["key"]
+    assert ("title", ["Hobbit"]) in results[0].match["fields"].items()
+
+
+def test_match(data):
```

Comment:
For sake of reader, might be nice to copy this raw entry inline to help follow the test - even if its obvious enough, given, ya know, LOTR hehe.

Reply:
It's a general problem with the current test suite. It relies quite a lot on the `test.bib` file, and is hard to read without looking at it all the time. We will have to do something about it at one point but prefer to deal with it altogether, I think...